pkg/users/local.js

```diff
@@ -1057,6 +1126,136 @@ function PageAccountConfirmDelete() {
     this._init();
 }
 
+function AccountExpiration() {
+    $("#account-expiration").on("show.bs.modal", function(ev) {
+        var account_id = $(ev.relatedTarget).data("account-id");
+        $("#account-expiration").data("account-id", account_id);
+
+        /* Fill in initial dialog values */
+        var expire_date = $(ev.relatedTarget).data("expire-date");
+        $('#account-expiration-expires').prop('checked', !!expire_date);
+        $('#account-expiration-date-input').val(expire_date ? expire_date.toLocaleDateString() : "");
+        $('#account-expiration-date-input').prop('disabled', !expire_date);
+    });
+
+    $('#account-expiration .btn-primary').on('click', function() {
+        var date, value;
+        var ex, promise = null;
+
+        /* Parse the dialog data and validate */
+        if ($('#account-expiration-expires').prop('checked')) {
+            value = $('#account-expiration-date-input').val();
+            if (!value) {
+                ex = new Error(_("Please specify an expiration date"));
+                ex.target = "#account-expiration-date-input";
+            } else {
+                date = new Date(value);
+                if (isNaN(date.getTime()) || date.getTime() < 0) {
+                    ex = new Error(_("Invalid expiration date"));
+                    ex.target = "#account-expiration-date-input";
+                }
+            }
+            if (ex)
+                promise = cockpit.reject(ex);
+        } else {
+            date = null;
+        }
+
+        /* Actually perform the action if valid */
+        var prog = ["/usr/sbin/usermod", "-e"];
+        var account_id = $("#account-expiration").data("account-id");
+        if (!promise) {
+            if (date)
+                prog.push(date.toISOString().substr(0, 10));
```

Comment:
This, in combination with `new Date` further up, depends on the Browser's time zone. I'm in CEST and the expiration date is always set to one day earlier than the day I entered in the dialog.
```js
(new Date('Jul 20, 2017')).toISOString().substr(0, 10)
// -> "2017-07-19"
```